src/mir_window.c

```diff
@@ -0,0 +1,542 @@
+        case KEY_LEFT:          return GLFW_KEY_LEFT;
+        case KEY_RIGHT:         return GLFW_KEY_RIGHT;
+        case KEY_DOWN:          return GLFW_KEY_DOWN;
+        case KEY_UP:            return GLFW_KEY_UP;
+        case KEY_F1:            return GLFW_KEY_F1;
+        case KEY_F2:            return GLFW_KEY_F2;
+        case KEY_F3:            return GLFW_KEY_F3;
+        case KEY_F4:            return GLFW_KEY_F4;
+        case KEY_F5:            return GLFW_KEY_F5;
+        case KEY_F6:            return GLFW_KEY_F6;
+        case KEY_F7:            return GLFW_KEY_F7;
+        case KEY_F8:            return GLFW_KEY_F8;
+        case KEY_F9:            return GLFW_KEY_F9;
+        case KEY_F10:           return GLFW_KEY_F10;
+        case KEY_F11:           return GLFW_KEY_F11;
+        case KEY_F12:           return GLFW_KEY_F12;
+        case KEY_F13:           return GLFW_KEY_F13;
+        case KEY_F14:           return GLFW_KEY_F14;
+        case KEY_F15:           return GLFW_KEY_F15;
+        case KEY_F16:           return GLFW_KEY_F16;
+        case KEY_F17:           return GLFW_KEY_F17;
+        case KEY_F18:           return GLFW_KEY_F18;
+        case KEY_F19:           return GLFW_KEY_F19;
+        case KEY_F20:           return GLFW_KEY_F20;
+        case KEY_F21:           return GLFW_KEY_F21;
+        case KEY_F22:           return GLFW_KEY_F22;
+        case KEY_F23:           return GLFW_KEY_F23;
+        case KEY_F24:           return GLFW_KEY_F24;
+        case KEY_KPSLASH:       return GLFW_KEY_KP_DIVIDE;
+        case KEY_KPDOT:         return GLFW_KEY_KP_MULTIPLY;
+        case KEY_KPMINUS:       return GLFW_KEY_KP_SUBTRACT;
+        case KEY_KPPLUS:        return GLFW_KEY_KP_ADD;
+        case KEY_KP0:           return GLFW_KEY_KP_0;
+        case KEY_KP1:           return GLFW_KEY_KP_1;
+        case KEY_KP2:           return GLFW_KEY_KP_2;
+        case KEY_KP3:           return GLFW_KEY_KP_3;
+        case KEY_KP4:           return GLFW_KEY_KP_4;
+        case KEY_KP5:           return GLFW_KEY_KP_5;
+        case KEY_KP6:           return GLFW_KEY_KP_6;
+        case KEY_KP7:           return GLFW_KEY_KP_7;
+        case KEY_KP8:           return GLFW_KEY_KP_8;
+        case KEY_KP9:           return GLFW_KEY_KP_9;
+        case KEY_KPCOMMA:       return GLFW_KEY_KP_DECIMAL;
+        case KEY_KPEQUAL:       return GLFW_KEY_KP_EQUAL;
+        case KEY_KPENTER:       return GLFW_KEY_KP_ENTER;
+        default:                return GLFW_KEY_UNKNOWN;
+    }
+}
+
+static void handleKeyEvent(MirKeyEvent const key, _GLFWwindow* window)
+{
+    int pressed = key.action == mir_key_action_up ?  GLFW_RELEASE : GLFW_PRESS;
+    int mods    = mirModToGLFWMod(key.modifiers);
+
+    long text = _glfwKeySym2Unicode(key.key_code);
+    int plain = !(mods & (GLFW_MOD_CONTROL | GLFW_MOD_ALT));
+
+    _glfwInputKey(window, toGLFWKeyCode(key.scan_code), key.scan_code, pressed, mods);
```

Comment:
I need to double check this... as I thought i checked the toGLFWKeyCode was working with the scan_code (with out the extra + 8). Maybe i just need to add it on to the second key.scan_code used?